lightning/src/ln/channelmanager.rs

```diff
@@ -3478,6 +3503,89 @@ impl<ChanSigner: ChannelKeys, M: Deref + Sync + Send, T: Deref + Sync + Send, K:
 	}
 }
 
+/// Used to signal to the ChannelManager persister that the manager needs to be re-persisted to
+/// disk/backups, through `block_until_needs_persist`.
+pub struct PersistenceNotifier {
+	/// Used to signal to the ChannelManager persister that the manager needs to be re-persisted to
+	/// disk/backups. Users won't access the persistence_lock directly, but rather wait on its bool
+	/// using block_until_needs_persist.
+	#[cfg(any(test, feature = "_test_utils"))]
+	pub persistence_lock: Arc<(Mutex<bool>, Condvar)>,
+	#[cfg(not(any(test, feature = "_test_utils")))]
+	persistence_lock: Arc<(Mutex<bool>, Condvar)>,
+}
+
+impl PersistenceNotifier {
+	fn new() -> Self {
+		Self {
+			persistence_lock: Arc::new((Mutex::new(false), Condvar::new())),
+		}
+	}
+
+	/// Used to signal to users when the ChannelManager needs to be persisted to disk. This function
+	/// blocks either for the given Duration or indefinitely until the ChannelManager has updates and
+	/// needs persistence. It returns a bool indicating whether persistence is necessary (which will
+	/// always be true if max_wait is None).
+	pub fn block_until_needs_persist(&self, max_wait: Option<Duration>) -> bool {
```

Comment:
This needs to be feature-gated as it would otherwise re-introduce dependence on current wall-clock time in the main `lightning` crate (which we finally removed a while back).